### PR TITLE
WIP : Add Action dynamicValue

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/OpenSRP/opensrp-plan-evaluator/issues/34

### CODE REVIEW

File: src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java [hunk 1]
```diff
@@ -45,6 +45,11 @@ public void generateTask(Resource resource, Action action, String planIdentifier
 		task.setExecutionEndDate(getDateTime(action.getTimingPeriod(), false));
 		task.setAuthoredOn(DateTime.now());
 		task.setLastModified(DateTime.now());
+        //TODO : Generate task according to defaultBusinessStatus
+//		if(action.getDynamicValue().getExpression().getName().equals("defaultBusinessStatus")) {
+//		task.setBusinessStatus();
+//
+//		}
```
Comment (githengi) [inline src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java:52]: Complete this implementation and add a unit test

File: src/main/java/org/smartregister/domain/Action.java [hunk 2]
```diff
@@ -41,6 +41,8 @@
 
     private String definitionUri;
 
+    private DynamicValue dynamicValue;
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/Action.java:44]: This should be a set of DynamicValue

File: src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java]: Done

File: src/main/java/org/smartregister/domain/Action.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/smartregister/domain/Action.java]: Updated

File: src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java [hunk 3]
```diff
@@ -45,7 +46,14 @@ public void generateTask(Resource resource, Action action, String planIdentifier
 		task.setExecutionEndDate(getDateTime(action.getTimingPeriod(), false));
 		task.setAuthoredOn(DateTime.now());
 		task.setLastModified(DateTime.now());
-		task.setBusinessStatus("Not Visited");
+		for (DynamicValue dynamicValue : action.getDynamicValue()) {
+			if (dynamicValue != null && dynamicValue.getExpression() != null &&
+					dynamicValue.getExpression().getName().equals("defaultBusinessStatus")) {
+				task.setBusinessStatus(dynamicValue.getExpression().getExpression());
+			} else {
+				task.setBusinessStatus("Not Visited");
```
Comment (githengi) [inline src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java:54]: This should be set after the for loop if task.getBusinessStatus is null

Comment (rehammuzzamil) [reply src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java]: Updated

**Pull request merged**